### PR TITLE
Fix kernel integration

### PR DESCRIPTION
**Pull request by bdunogier**
> [EZP-26982](https://jira.ez.no/browse/EZP-26982)

Fixes the integration of the package into ezpublish-kernel.

### Tasks
- [x] Make `AppCache` use `TagAwareStore`
- [x] Make sure that purge requests use the `key` header and not `xkey`
- [x] Write a specification for `KernelPass`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,13 @@ Add the package to `app/AppKernel.php`, *before* the EzPublishCoreBundle declara
 
 The package will replace the services from the kernel, thus enabling the new features, such as multi-tagging.
 
+The application cache class needs to be customized. You can do so by setting the `SYMFONY_HTTP_CACHE_CLASS`
+environment variable:
+
+    export SYMFONY_HTTP_CACHE_CLASS='EzSystems\PlatformHttpCacheBundle\AppCache'
+
+Do not forget to restart your web server.
```
Comment (andrerom) [inline README.md]: alternatively user needs to extend it, for use cases where they need to add own customizations.

Comment (bdunogier) [inline README.md]: How would you phrase that ? I expect that somebody who has extended it knows what needs to be done at this point.

Comment (glye) [inline README.md]: Keep it simple: "Alternatively, if you need to make your own customizations, you should extend this class and set `SYMFONY_HTTP_CACHE_CLASS` accordingly."

Comment (andrerom) [inline README.md]: "and set SYMFONY_HTTP_CACHE_CLASS accordingly."

if you extend class from this package you don't need to set the env variable, as you'll use the default AppCache class _(which will be the one to extend the class in this package)_.

File: src/DependencyInjection/Compiler/KernelPass.php [hunk 2]
```diff
@@ -9,18 +9,14 @@
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 
-/**
- *
```
Comment (glye) [inline src/DependencyInjection/Compiler/KernelPass.php]: Could have added some doc instead ;)

Comment (bdunogier) [inline src/DependencyInjection/Compiler/KernelPass.php]: Docs added before merging :)

**Pull request merged**